wecs/mechanics/clock.py

```diff
@@ -93,20 +96,24 @@ class UpdateCalendar(System):
         ]),
     }
 
+    def increment_timeframes(self, calendar, time_since):
+        timeframes = list(calendar.timeframes.keys())
```

Comment:
...and here's the problem with dicts, this list will effectively in random order. Althouth there is also OrderedDict, so it doesn't have to be a list.